patches/MCLOUD-3025__sitemap_generation_warnings__2.3.3.patch

```diff
@@ -0,0 +1,48 @@
+diff -Nuar a/vendor/magento/module-sitemap/Model/Observer.php b/vendor/magento/module-sitemap/Model/Observer.php
+--- a/vendor/magento/module-sitemap/Model/Observer.php
++++ b/vendor/magento/module-sitemap/Model/Observer.php
+@@ -61,6 +61,11 @@ class Observer
+      */
+     private $emailNotification;
+ 
++    /**
++     * @var int
++     */
++    private $retryInterval = 10;
++
+     /**
+      * Observer constructor.
+      * @param ScopeConfigInterface $scopeConfig
+@@ -87,6 +92,7 @@ class Observer
+     public function scheduledGenerateSitemaps()
+     {
+         $errors = [];
++        $sitemapsWithError = [];
+         $recipient = $this->scopeConfig->getValue(
+             Observer::XML_PATH_ERROR_RECIPIENT,
+             ScopeInterface::SCOPE_STORE
+@@ -107,11 +113,22 @@ class Observer
+             try {
+                 $sitemap->generateXml();
+             } catch (\Exception $e) {
++                $sitemapsWithError[] = $sitemap;
+                 $errors[] = $e->getMessage();
+             }
+         }
+-        if ($errors && $recipient) {
+-            $this->emailNotification->sendErrors($errors);
++        if ($errors && $recipient && !empty($sitemapsWithError)) {
++            sleep($this->retryInterval);
++            $nonRecoverableErrors = [];
++            //re-try to write sitemap again
++            foreach ($sitemapsWithError as $sitemap) {
++                try {
++                    $sitemap->generateXml();
++                } catch (\Exception $e) {
++                    $nonRecoverableErrors[] = $e->getMessage();
++                }
++            }
++            $this->emailNotification->sendErrors($nonRecoverableErrors);
+         }
+     }
+ }
```

Comment:
add empty line

Reply:
Are you sure that empty line is needed at the patch not at the end of the php file?

Reply:
Yes, I am sure. Actually, it should be added by default when you create patch, so it is strange that you do not have it.

Reply:
I have it? but it's not reflected in review, as well as in any other reviews for patches:

Reply:
https://github.com/magento/magento-cloud-patches/pull/35/files
https://github.com/magento/magento-cloud-patches/pull/34/files
https://github.com/magento/magento-cloud-patches/pull/32/files

Reply:
Extra empty line at the end is not displayed in the PR, but if you don't have, you will see that red mark as you had before in this file.